specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2023-01-15-preview/failoverGroups.json

```diff
@@ -0,0 +1,231 @@
+          "type": "string",
+          "enum": [
+            "async",
+            "sync"
+          ],
+          "x-ms-enum": {
+            "name": "failoverGroupPartnerSyncMode",
+            "modelAsString": true
+          },
+          "default": "async",
+          "description": "The partner sync mode of the SQL managed instance."
+        },
+        "role": {
+          "type": "string",
+          "enum": [
+            "primary",
+            "secondary",
+            "force-primary-allow-data-loss",
+            "force-secondary"
+          ],
+          "x-ms-enum": {
+            "name": "instanceFailoverGroupRole",
+            "modelAsString": true
+          },
+          "default": "primary",
+          "description": "The role of the SQL managed instance in this failover group."
+        }
+      },
+      "additionalProperties": {},
+      "required": [
+        "role"
+      ]
+    },
+    "FailoverGroupStatus": {
+      "type": "object",
+      "description": "The status of the Kubernetes custom resource.",
+      "properties": {
+        "lastUpdateTime": {
+          "type": "string",
+          "description": "The time that the custom resource was last updated."
+        },
+        "observedGeneration": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The version of the replicaSet associated with the failover group custom resource."
+        },
+        "state": {
+          "type": "string",
+          "description": "The state of the failover group custom resource."
+        },
+        "results": {
+          "type": "string",
+          "description": "The message in case of a failure in the failover group."
+        },
+        "role": {
+          "type": "string",
+          "description": "The role of the managed instance."
+        },
+        "replicas": {
+          "type": "array",
+          "description": "A list of failover group replicas.",
+          "items": {
+            "$ref": "#/definitions/AGReplicas",
+            "description": "null",
+            "x-ms-client-flatten": false
+          },
+          "x-ms-identifiers": [
+            "replicaName"
+          ]
+        }
+      },
+      "additionalProperties": {}
+    },
+    "AGReplicas": {
+      "type": "object",
+      "description": "The information about each failover group replica.",
+      "properties": {
+        "healthState": {
+          "type": "string",
+          "description": "The health state of the replica."
+        },
+        "replicaName": {
+          "type": "string",
+          "description": "The replica name."
+        },
+        "role": {
+          "type": "string",
+          "description": "The role of the replica."
+        },
+        "connectedState": {
+          "type": "string",
+          "description": "The connected state."
+        },
+        "synchronizationState": {
+          "type": "string",
+          "description": "The synchronization state."
```

Comment:
>The synchronization state [](http://example.com/codeflow?start=26&length=25)

Can you provide better descriptions everywhere?

Reply:
Sure, I updated the descriptions where I could.